pkg/scanners/scannerv4/convert.go

```diff
@@ -166,7 +166,6 @@ func vulnerabilities(vulnerabilities map[string]*v4.VulnerabilityReport_Vulnerab
 			Cve:      ccVuln.GetName(),
 			Advisory: advisory(ccVuln.GetAdvisory()),
 			Summary:  ccVuln.GetDescription(),
-			// TODO(ROX-26547)
 			// The link field will be overwritten if preferred CVSS source is available
 			Link:        link(ccVuln.GetLink()),
```

Comment:
Can we set the `vuln.Link` on line 180, guarded by a `if len(vuln.CVSSMetrics) != 0`? The the logic for setting the link is in the same place and the comment above is unnecessary.

Reply:
I added this change to method 'setScoresAndScoreVersions()', it's also guarded by CVSSMetrics length check. Also there are tests around 'setScoresAndScoreVersions()' so I think that was a better approach. Please let know your thoughts.

Reply:
I know I didn't remove `ccVuln.GetLink()` but we only use that value in case CVSS metrics is missing. Do you think that's acceptable?

Reply:
> Please let know your thoughts.

In that case don't set the `Link` in line 170, pass `ccVuln.GetLink()` to `setScoresAndScoreVersions()`, set the link field using `ccVuln.GetLink()` inside `setScoresAndScoreVersions`, and add test cases to `setScoresAndScoreVersions` to cover the scenarios where it should be overwritten or not overwritten.